scripts/nbconverted/integrate-msigdb.py

```diff
@@ -0,0 +1,398 @@
+)
+
+coding_genes = set(gene_df['GeneID'])
+print(gene_df.shape)
+gene_df.head(2)
+
+
+# In[9]:
+
+
+# Subset the msigdb genes to the gene curation above
+common_subset_genes = set(gene_df['Symbol']).intersection(set(msigdb_df.index))
+msigdb_subset_genes_df = msigdb_df.loc[common_subset_genes, :]
+
+# What is the new distribution of pathway sizes?
+msigdb_subset_genes_df.sum(axis=1).sort_values().hist(bins=30);
+
+
+# In[10]:
+
+
+# What is the new distribution of gene representation?
+msigdb_subset_genes_df.sum(axis=0).sort_values().hist(bins=30);
+
+
+# In[11]:
+
+
+# The genes that were removed from MSigDB participate mostly in few pathways
+diff_genes = set(msigdb_df.index).difference(gene_df['Symbol'])
+msigdb_other_genes_df = msigdb_df.loc[diff_genes, :]
+msigdb_other_genes_df.sum(axis=1).sort_values().hist(bins=30)
+
+
+# ## Add genes as nodes to the graph
+# 
+# Use the gene-symbol identifier for easier interpretation
+
+# In[12]:
+
+
+get_ipython().run_cell_magic('time', '', "for i, row in gene_df.iterrows():\n    # Build dictionary of descriptive elements for each gene\n    meta_data = {\n        'description': row['description'],\n        'source': 'Entrez Gene',\n        'url': 'http://identifiers.org/ncbigene/{}'.format(row['GeneID']),\n        'license': 'CC0 1.0',\n    }\n    \n    if pd.notnull(row['chromosome']):\n        meta_data['chromosome'] = row['chromosome']\n\n    # Add genes to graph\n    graph.add_node(kind='Gene', identifier=row['Symbol'], name=row['GeneID'],\n                   data=meta_data)")
+
+
+# ## Add gene set nodes and associated genes as edges
+# 
+# Add each MSigDB collection as distinct nodes with a `participates` edge for representative gene sets and corresponding membership.
+
+# In[13]:
+
+
+def add_msigdb_node_to_graph(current_graph, collection_file, collection_kind,
+                             collection_source, gene_list, min_geneset_size=4,
+                             max_geneset_size=1000, license='CC BY 4.0'):
+    """
+    Add nodes and edges to current graph based on geneset memembership of collection
+    
+    Arguments:
+    current_graph - a hetnet object to add node-edge info to
+    collection_file - location of msigdb file
+    collection_kind - the kind of node already initialized in the graph
+    collection_source - alternative ID for collection
+    gene_list - a list of genes to consider when building the graph
+    min_geneset_size - filter out a given gene set if it has fewer genes
+    max_geneset_size - filter out a given gene set if it has more genes
+    license - given license associated with node
+    
+    Output:
+    Adds to current graph; Returns the amount of filtered genesets
+    """
+    
+    # Build meta data dictionary to store node info
+    meta_data = {'license': license, 'source': collection_source}
+    
+    # Open the .gmt file and process each geneset
+    filtered_genesets = []
+    with open(collection_file, 'r') as collection_fh:
```

Comment:
Is nice to separate out GMT reading into a generative fxn like:

```python
def read_gmt(path):
    """Generate rows from a .gmt file."""
    read_file = open(path)
    reader = csv.reader(read_file, delimiter='\t')
    for row in reader:
        name = row[0]
        description = row[1]
        genes = set(row[2:])
        yield name, description, genes
    read_file.close()
```

But change that fxn to use `with` for file open. Not sure what I was thinking not using context management.

Reply:
I think its easier to read the way I have it. It also takes only seconds to run